lib/Conversion/TritonToStructured/TritonToStructuredPass.cpp

```diff
@@ -90,8 +90,9 @@ class TritonToStructuredPass
       // We only care about tensor of index / int (in addition to pointer type)
       // because only values of int and index type can potentially be part of a
       // pointer arithmetic sequence.
-      if (!isa<triton::PointerType>(tensorType.getElementType()) &&
-          !tensorType.getElementType().isIntOrIndex()) {
+      auto elementType = tensorType.getElementType();
+      if (isa<triton::PointerType>(elementType) ||
+          (elementType.isIntOrIndex() && !elementType.isInteger(1))) {
```

Comment:
Do we need to reverse the condition?
If we have to change if and else,  need to move the comment about std::nullopt as well.